### PR TITLE
Only activate "Show detail" pan gesture when swiping left

### PR DESCRIPTION
**Pull request by michaelkirk**
Fixes https://github.com/WhisperSystems/Signal-iOS/issues/2879

PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/views/DirectionalPanGestureRecognizer.swift [hunk 1]
```diff
@@ -17,19 +16,50 @@ class PanDirectionGestureRecognizer: UIPanGestureRecognizer {
...
+            let deltaX = previousLocation.x - location.x
+
+            switch direction {
+            case .down where deltaY < 0:
+                return
+            case .up where deltaY > 0:
+                return
+            case .forward where deltaX < 0:
+                return
+            case .backward where deltaX > 0:
```
Comment (charlesmchen) [inline Signal/src/views/DirectionalPanGestureRecognizer.swift]: What about RTL?

File: Signal/src/views/DirectionalPanGestureRecognizer.swift [hunk 2]
```diff
@@ -17,19 +16,50 @@ class PanDirectionGestureRecognizer: UIPanGestureRecognizer {
...
         if state == .began {
             let vel = velocity(in: view)
             switch direction {
-            case .horizontal where fabs(vel.y) > fabs(vel.x):
-                state = .cancelled
-            case .vertical where fabs(vel.x) > fabs(vel.y):
-                state = .cancelled
+            case .forward, .backward:
+                if fabs(vel.y) > fabs(vel.x) {
+                    state = .cancelled
```
Comment (charlesmchen) [inline Signal/src/views/DirectionalPanGestureRecognizer.swift]: `UIGestureRecognizer`s can't always be cancelled; IMO safer to have an `abortGesture()` method that not only _tries_ to enter cancelled state but also toggles the enabled/disabled state, thereby forcing the gesture to abort.

Comment (michaelkirk) [reply Signal/src/views/DirectionalPanGestureRecognizer.swift]: When can't they be cancelled?

File: Signal/src/views/DirectionalPanGestureRecognizer.swift (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/views/DirectionalPanGestureRecognizer.swift]: Good call, thanks. Updated!

PTAL @charlesmchen

File: Signal/src/views/DirectionalPanGestureRecognizer.swift (hunk 2 above)
Comment (charlesmchen) [reply Signal/src/views/DirectionalPanGestureRecognizer.swift]: It's a particularity of `UIGestureRecognizer` that cancelling doesn't always work.  I can't remember the exact repro, but it's not terribly obscure.

**Pull request merged**